### PR TITLE
Render the angular component as the root element of the react component

### PR DESCRIPTION
**Pull request by jrsquared**
In our conversion story, all of our angular components are already wrapped in a single component and adding the wrapper div breaks a lot of our existing layout. Fits our model better to be able to just replace.

Also added tests around this new flag.

### CODE REVIEW

**Comment by bcherny** [comment]
@jrsquared Thanks for the contribution! A few notes:

1. Let's make this the default behavior - no need for `replace` flag
2. Can you add a test to make sure that the react `div` is properly unmounted from the DOM when the component is destroyed?

**Comment by jrsquared** [comment]
1. Done. Regarding #2, it actually wasn't calling $destroy because we were losing the react context on replace. So I cleaned it up and made the angular component actually be the root element of the react component.
2. Added the test.

File: test.tsx [hunk 1]
```diff
@@ -31,11 +30,13 @@ const FooBarBaz = {
   },
   controller: FooController,
   template: `
-    <span>{{ $ctrl.foo }}</span>
-    <span>{{ $ctrl.fooBar*12 }}</span>
-    <span>{{ $ctrl.fooService.add42($ctrl.fooBar) }}</span>
-    <div ng-click="$ctrl.onClick()">Click Me</div>
-    <ng-transclude></ng-transclude>
+    <span id="foobarbaz">
```
Comment (bcherny) [inline test.tsx]: With this PR, do component templates now need to be wrapped in an enclosing element?

File: test.tsx [hunk 2]
```diff
@@ -64,65 +65,77 @@ it('should give a react component', () => {
...
-  expect($(element).find('span').eq(1).text()).toBe('516')
+  expect($(element).find('span').eq(2).text()).toBe('516')
   unmountComponentAtNode(element)
 })
 
 it('should destroy', () => {
   const Foo2 = compile($injector)
   const element = document.createElement('div')
-  render(<Foo2 foo='hello' fooBar={42} />, element)
+  render(<Foo2 foo='hello' fooBar={42} />, element) as Element
```
Comment (bcherny) [inline test.tsx]: This assert doesn't seem to be used

File: test.tsx [hunk 3]
```diff
@@ -64,65 +65,77 @@ it('should give a react component', () => {
...
   spyOn(FooController.prototype, '$onDestroy')
-  unmountComponentAtNode(element)
+  const unmounted = unmountComponentAtNode(element)
+  expect(unmounted).toBeTruthy
```
Comment (bcherny) [inline test.tsx]: Typo: Be sure to invoke `toBeTruthy`

Comment (bcherny) [inline test.tsx]: Since we're doing more magic now, can you also assert that `element` is now empty?

File: test.tsx [hunk 4]
```diff
@@ -64,65 +65,77 @@ it('should give a react component', () => {
...
+  unmountComponentAtNode(element)
 })
 
-function compile($injector: auto.IInjectorService) {
-  interface Props {
-    bazMoo1Boo?(value: number): any
-    foo: string
-    fooBar: number
-  }
+it('should use the angular component as the root component', () => {
```
Comment (bcherny) [inline test.tsx]: Nice

File: test.tsx (hunk 1 above)
Comment (jrsquared) [inline test.tsx]: Actually, now that I switched to just use the angular component as the root, probably not. I'll take a look at this, good point.

File: test.tsx (hunk 2 above)
Comment (jrsquared) [inline test.tsx]: Yep, cruft, good catch.

**Comment by jrsquared** [comment]
Addressed all comments.

**Comment by bcherny** [comment]
Thanks for the contribution @jrsquared! Published this change @2.0.0

**Pull request merged**